src/pg_to_evalscript/node.py

```diff
@@ -266,3 +267,33 @@ def write_process(self):
     return count(arguments);
 }}
 """
+
+class ArrayApplyNode(Node):
+    def is_process_defined(self, process_id):
+        return True
+        
+    def write_process(self):
+        newline = "\n"
+        tab = "\t"
+        return f"""
+function array_apply(arguments) {{
+    function process(arguments) {{
+    {newline.join(node.write_function() for node in self.child_nodes)}
+    {newline.join(node.write_call() for node in self.child_nodes)}
+        return {self.child_nodes[-1].node_varname_prefix + self.child_nodes[-1].node_id};
+    }}
+
+    const {{data, context = null}} = arguments;
```

Comment:
I think we should check if `data` is an array.

Reply:
The same long error stack is printed for unit tests if you put a test for errors in the method that checks valid inputs.
Tests for errors should be added (similar to the ones in unit tests), so that the error is checked.

These tests should be added:
- `data` is not array
- `data` is empty array (`[]`) - currently returns `[null]` but should return empty array (`[]`) as per docs
  > Return Value
  >
  > An array with the newly computed values. The number of elements are the same as for the original array.
- element in `data` is not appropriate for the applied process (e.g. string and `add` process)


Maybe something like this for the errors that the applied process could ~~return~~ throw:

```
    for (let i = 0; i < data.length; i++) {{
        try{{
          newData[i] = process({{
              x: data[i],
              index: i,
              label: data.labels ? data.labels[i] : undefined,
              context: context,
          }});
        }}
        catch (error_from_the_applied_process) {{
          // propagate that same error
          // or throw a "wrapper" error adding the info 
          // for which parent process and which element
          // the error in the applied process happened
        }}
    }}
```

Reply:
Added all these additional test cases and a few more. Can you explain please how you got to the result of the second test case that you mentioned, where data is an empty array `[]`. I can't understand how this could happen, since the for loop runs "through" the data array and if it is empty, nothing should happen and an empty array should be returned.

Regarding the long stack trace, nothing changes whether I simply catch and propagate the same error or create a new error or actually just catch the error and don't do anything with it. In all cases the same long stack trace remains.

Reply:
> Can you explain please how you got to the result of the second test case that you mentioned, where data is an empty array []. I can't understand how this could happen, since the for loop runs "through" the data array and if it is empty, nothing should happen and an empty array should be returned.

I tried to get to the bottom of this but I couldn't find where the `null` (or `None` as python interprets it) comes from, so I think we can leave that to be solved later.

> Regarding the long stack trace, nothing changes whether I simply catch and propagate the same error or create a new error or actually just catch the error and don't do anything with it. In all cases the same long stack trace remains.

I think the long stack trace is correct so I think it's just the matter of catching (or not) the error in this function, which is more to help the developers / users of this project than anything else.